core/templates/components/skill-mastery/skill-mastery.component.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the skill mastery viewer.
+ */
+
+import { SkillMasteryListConstants } from
+  'components/skills-mastery-list/skills-mastery-list.constants';
+
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { SkillMasteryBackendApiService } from
+  'domain/skill/skill-mastery-backend-api.service';
+
+@Component({
+  selector: 'skill-mastery-viewer',
+  templateUrl: './skill-mastery.directive.html',
+  styleUrls: []
+})
+export class SkillMasteryViewerComponent implements OnInit {
+  @Input() skillId: string;
+  @Input() masteryChange: number;
+
+  skillMasteryDegree: number = 0;
+
+  constructor(
+    private skillMasteryBackendApiService: SkillMasteryBackendApiService
+  ) {}
+
+  ngOnInit() {
+    this.skillMasteryDegree = 0.0;
+
+    this.skillMasteryBackendApiService.fetchSkillMasteryDegrees(
+      [this.skillId]).then(function(degreesOfMastery) {
+      this.skillMasteryDegree = degreesOfMastery[this.skillId];
+    });
```

Comment:
Use arrow functions. Otherwise, you might end up losing the context of "this".
```suggestion
      [this.skillId]).then(degreesOfMastery => this.skillMasteryDegree = (
        degreesOfMastery[this.skillId]));
```

Reply:
Done, but I don't understand why I'd end up losing the context of this. Is it because the scope would change?

Reply:
In classic function expressions, the "this" keyword is bound to different values based on the context in which it is called. With arrow functions, "this" is lexically bound. It means that it uses "this" from the code that contains the arrow function. So changing all your from "function(args) {}" to "(args) => {}" will solve your problem. In case you don't like arrow function and prefer to use function() {}, you can just add ".bind(this)" to the end of the function so that it binds the value of this to the function. So it will look like this "function(args) {}.bind(this);"

Reply:
Hmm, okay, that makes sense. Thanks!